components/search/src/filter.rs

```diff
@@ -56,10 +59,15 @@ pub(crate) fn filter_engine_configuration(
     let mut default_engine_id: Option<String> = None;
     let mut default_private_engine_id: Option<String> = None;
 
+    let mut user_environment = user_environment.clone();
+    user_environment.locale = user_environment.locale.to_lowercase();
+    user_environment.region = user_environment.region.to_lowercase();
+    user_environment.version = user_environment.version.to_lowercase();
+
```

Comment:
The application name is passed in [as an enumeration](https://github.com/mozilla/application-services/blob/ab36d7adceab85bcc344d117d6bcb7a3e0b7097d/components/search/src/types.rs#L9-L29) rather than a string, hence we don't need to lower case it here.